components/Modals/BatchTransfer/index.tsx

```diff
@@ -0,0 +1,343 @@
+                            progress: undefined,
+                            theme: toastTheme(isLight),
+                            toastId: 2,
+                        })
+                        throw Error
+                    }
+                }
+                for (let i = 0; i < inputCoin.length; i += 1) {
+                    const amount = inputCoin[i]?.split('@')[1]
+                    const input_ = Number(amount)
+                    if (isNaN(input_)) {
+                        toast.error(t('The input is not a number.'), {
+                            position: 'bottom-right',
+                            autoClose: 2000,
+                            hideProgressBar: false,
+                            closeOnClick: true,
+                            pauseOnHover: true,
+                            draggable: true,
+                            progress: undefined,
+                            theme: toastTheme(isLight),
+                            toastId: 3,
+                        })
+                        throw new Error()
+                    }
+                }
+                setSavedCurrency(true)
+            } catch {
+                setSavedCurrency(false)
+            }
+            console.log(inputCoin)
+        }
+    }
+
+    const resetState = () => {
+        setInputCoin([])
+        setSelectedCoin([])
+        setSavedCurrency(false)
+    }
+
+    const handleSubmit = async () => {
+        setIsLoading(true)
+        const zilpay = new ZilPayBase()
+        let params: any = []
+        const addr: TransitionParams = {
+            vname: 'addr',
+            type: 'ByStr20',
+            value: resolvedInfo?.addr,
+        }
+        params.push(addr)
+        let arrayToken: any = []
+        for (let i = 0; i < inputCoin.length; i += 1) {
+            const val = inputCoin[i].split('@')
+            console.log(inputCoin[i])
+            const _currency = tyron.Currency.default.tyron(
+                val[0],
+                Number(val[1])
+            )
+            arrayToken.push({
+                argtypes: ['String', 'Uint128'],
+                arguments: [`${val[0]}`, `${_currency.amount}`],
+                constructor: 'Pair',
+            })
+        }
+        const tokens: TransitionParams = {
+            vname: 'tokens',
+            type: 'List( Pair String Uint128 )',
+            value: arrayToken,
+        }
+        params.push(tokens)
+        const tyron_ = await tyron.Donation.default.tyron(donation!)
+        const tyron__: TransitionParams = {
+            vname: 'tyron',
+            type: 'Option Uint128',
+            value: tyron_,
+        }
+        params.push(tyron__)
+
+        dispatch(setTxStatusLoading('true'))
+        updateModalTxMinimized(false)
+        updateModalTx(true)
+        let tx = await tyron.Init.default.transaction(net)
+        await zilpay
+            .call({
+                contractAddress: resolvedInfo?.addr!,
```

Comment:
this is valid only when withdrawing, otherwise it should be the originator address.

Reply:
done